### PR TITLE
Don't incorrectly skip the multiblock test

### PR DESCRIPTION
**Pull request by mattcaswell**
sslapitest has a test for multiblock writes. Since multiblock writing is
only available on some platforms the multiblock test checks whether we are
on such a platform first, and skips the test if we are not. Unfortunately
a bug in the check meant that we always skipped the test regardless of the
platform.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]
Looks good.  I think the ubsan fix should be done here but would be okay with doing it separately.

**Comment by mattcaswell** [comment]
Fix pushed for ubsan issue. CIs are still in progress but the ubsan build has already passed.

@paulidale - please can you reconfirm?

Ping @openssl/committers for second review.

**Comment by mattcaswell** [comment]
All CIs are green.

**Review by beldmit — APPROVED** [review]
LGTM

**Review by paulidale — APPROVED** [review]

**Comment by hlandau** [comment]
Merged to master and 3.0. Thank you.

**Pull request closed without merging**
